src/extension/module.js

```diff
@@ -1139,11 +1143,38 @@
               return;
             }
           }
-          updatePreview();
+          if (location.pathname.startsWith('/:x:/')) {
+            // refresh excel with preview param
+            window.sessionStorage.setItem('hlx-sk-preview', JSON.stringify({
+              previewUrl: location.href,
+              previewTimestamp: Date.now(),
+            }));
+            window.location.reload();
+          } else {
+            updatePreview(sk);
+          }
         },
         isEnabled: (sidekick) => sidekick.isAuthorized('preview', 'write')
           && sidekick.status.webPath,
       },
+      callback: () => {
+        const { location } = sk;
+        const { previewUrl, previewTimestamp } = JSON
+          .parse(window.sessionStorage.getItem('hlx-sk-preview') || '{}');
+        window.sessionStorage.removeItem('hlx-sk-preview');
+        if (previewUrl === location.href && previewTimestamp < Date.now() + 60000) {
```

Comment:
isn't that a bit long?

Reply:
Do you see any harm in it? I didn't want to cut it too close, in case of a slow connection... I think the fact that it has to be in session storage plus within the next minute makes it pretty safe...